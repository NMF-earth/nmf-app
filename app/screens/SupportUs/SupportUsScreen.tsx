import React from "react";
import { ScrollView } from "react-native";

import { Text } from "../../components";
import styles from "./SupportUsScreen.styles";
import { t } from "../../utils";
import navigationOptions from "./SupportUsScreen.navigationOptions";

const SupportUsScreen = () => (
  <ScrollView style={styles.container}>
    <Text.H2 style={styles.title}>
      {t("SUPPORT_US_SCREEN_WHY_DONATE_TITLE")}
    </Text.H2>
    <Text.Primary style={styles.paragraph}>
      {t("SUPPORT_US_SCREEN_WHY_DONATE_CONTENT")}
    </Text.Primary>
    <Text.Primary style={styles.paragraph}>
      {t("SUPPORT_US_SCREEN_TITLE_CONSIDER_DONATE")}
    </Text.Primary>
    <Text.Primary style={styles.paragraph}>
      {t("SUPPORT_US_SCREEN_PLATEFORM")}
    </Text.Primary>
    <Text.Link url="https://www.patreon.com/nmf_earth" style={styles.paragraph}>
      Patreon
    </Text.Link>
    <Text.Link
      url="https://opencollective.com/nmf_earth"
      style={styles.paragraph}
    >
      Open Collective
    </Text.Link>
  </ScrollView>
);

SupportUsScreen.navigationOptions = navigationOptions;

export default SupportUsScreen;
