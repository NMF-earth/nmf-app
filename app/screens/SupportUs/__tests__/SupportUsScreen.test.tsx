import React from "react";
import renderer from "react-test-renderer";
import SupportUsScreen from "../SupportUsScreen";

it("SupportUsScreen renders correctly", () => {
  const tree = renderer.create(<SupportUsScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
