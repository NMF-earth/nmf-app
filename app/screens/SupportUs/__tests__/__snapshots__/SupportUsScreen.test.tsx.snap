// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SupportUsScreen renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "flex": 1,
      "paddingHorizontal": 14,
    }
  }
>
  <View>
    <Text.H2
      style={
        Object {
          "paddingBottom": 4,
          "paddingTop": 14,
        }
      }
    >
      Why you should consider supporting us
    </Text.H2>
    <Text.Primary
      style={
        Object {
          "paddingVertical": 10,
        }
      }
    >
      NMF.earth is not for profit, we have only raised a small amount on Kickstarter and we keep working a lot on the project even if it's not financially sustainable. Besides that, we have many costs such as servers, domain name, softwares licences, apple developer account etc.
    </Text.Primary>
    <Text.Primary
      style={
        Object {
          "paddingVertical": 10,
        }
      }
    >
      If you like this project, please consider donating a small amount to help us keep going! And since everything is transparent with us, you will be able to see how we used your donation by looking at end of the year report.
    </Text.Primary>
    <Text.Primary
      style={
        Object {
          "paddingVertical": 10,
        }
      }
    >
      You can do so on Open Collective or on Patreon.
    </Text.Primary>
    <Text.Link
      style={
        Object {
          "paddingVertical": 10,
        }
      }
      url="https://www.patreon.com/nmf_earth"
    >
      Patreon
    </Text.Link>
    <Text.Link
      style={
        Object {
          "paddingVertical": 10,
        }
      }
      url="https://opencollective.com/nmf_earth"
    >
      Open Collective
    </Text.Link>
  </View>
</RCTScrollView>
`;
